lib/mapnik_backend.js

```diff
@@ -175,29 +166,21 @@ MapnikSource.prototype._open = function(uri) {
 
     // Initialize this map. This wraps `localize()` and calls `create()`
     // afterwards to actually create a new Mapnik map object.
-    Step(function() {
-        source._loadXML(this);
-    }, function(err, xml) {
-        if (err) return error(err);
-        // https://github.com/mapbox/tilelive-mapnik/issues/25
-        // there seems to be no value to assinging xml to a property
-        //source._xml = xml;
-        source._createMetatileCache(uri.query.metatileCache);
-        source._createPool(xml, this);
-    }, function(err) {
-        if (err) return error(err);
-        source._populateInfo(this);
-    }, function(err) {
-        if (err) return error(err);
-        source.open = true;
-        source.emit('open', null, source);
+    this._loadXML((err, xml) => {
+        if (err) return this.emit('open', err);
+
+        this._createMetatileCache(uri.query.metatileCache);
+        this._createPool(xml);
+        this._populateInfo((err_populate) => {
```

Comment:
Careful!!. You've just removed this callback. See https://github.com/CartoDB/tilelive-mapnik/pull/23/files#diff-92989f54c4804505d7ebdacce2a1a9b9R245